src/toast/ops/conviqt.py

```diff
@@ -732,26 +747,28 @@ def _exec(self, data, detectors=None, **kwargs):
             beam_T, beam_P= self.get_beam(beam_file, det, verbose)
 
             detector = self.get_detector(det)
-            
-            
-            theta, phi, psi, psi_pol = self.get_pointing(data, det, verbose)
+
+            theta, phi, psi, psi_pol, hwp_angle = self.get_pointing(data, det, verbose)
             # T-convolution
             pnt = self.get_buffer(theta, phi, psi, det, verbose)
 
-            convolved_data = self.convolve(skyT, beam_T, detector, pnt, det, verbose)
+            convolved_data = self.convolve(
+                skyT, beam_T, detector, pnt, det, verbose, pol=False
+            )
 
             del (pnt,)
-            # E-convolution
+            # EB-convolution
             pnt = self.get_buffer(theta, phi, psi, det, verbose)
-            convolved_data += np.cos(2 * psi_pol) * self.convolve(
-                skyE, beam_P, detector, pnt, det, verbose
+            convolved_data += np.cos(4 * hwp_angle) * self.convolve(
+                skyEB, beam_P, detector, pnt, det, verbose, pol=True
             )
             del (pnt,)
-            # B-convolution
+            # BE-convolution
             pnt = self.get_buffer(theta, phi, psi, det, verbose)
-            convolved_data += np.sin(2 * psi_pol) * self.convolve(
-                skyB, beam_P, detector, pnt, det, verbose
+            convolved_data += np.sin(4 * hwp_angle) * self.convolve(
```

Comment:
It is not clear  to me why now the convolved data are weighted by the sin(or cosine) of the hwp angle and not by the psi_pol angle..

Reply:
The psi_pol angle is already included in the beam and pointing matrix